source/pdo_sqlsrv/pdo_dbh.cpp

```diff
@@ -1463,7 +1463,7 @@ int pdo_sqlsrv_dbh_quote( _Inout_ pdo_dbh_t* dbh, _In_reads_(unquoted_len) const
         if ( encoding == SQLSRV_ENCODING_UTF8 ) {
             quotes_needed = 3;
         }
-        for ( size_t index = 0; index < unquoted_len && unquoted[ index ] != '\0'; ++index ) {
+        for ( size_t index = 0; index < unquoted_len; ++index ) {
```

Comment:
Is the argument `unquoted_len` always correct? What is `unquoted_len` if the string is empty? Or what if the input string is composed of only one null character, something like `'\0'`, or a series of null characters, something like `'\0\0\0\0'`?

Reply:
From what I've tested, unquote_len seems to be always correct ( it was parsed out by the PDO extension using zend_parse_parameter. If the string is empty (""), unquoted_len is 0. If the input string is composed of only one null character, unquoted_len is 1, and for '\0\0\0\0' unquoted_len is 4.